src/impl/tpls/KokkosSparse_spmv_tpl_spec_avail.hpp

```diff
@@ -116,6 +116,21 @@ struct spmv_tpl_spec_avail<const SCALAR, const ORDINAL, Kokkos::Device<Kokkos::C
 #endif  // CUDA/CUSPARSE >= 9.0?
 #endif  // KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
 
+// cuSPARSE
+#ifdef KOKKOSKERNELS_ENABLE_TPL_ROCSPARSE
+
+//These versions of cuSPARSE require the ordinal and offset types to be the same.
+//For KokkosKernels, this means int/int only.
+
+#define KOKKOSSPARSE_SPMV_TPL_SPEC_AVAIL_ROCSPARSE(SCALAR, ORDINAL, OFFSET, XL, YL, MEMSPACE) \
+template <> \
+struct spmv_tpl_spec_avail<const SCALAR, const ORDINAL, Kokkos::Device<Kokkos::Cuda, MEMSPACE>, Kokkos::MemoryTraits<Kokkos::Unmanaged>, const OFFSET, \
+			   const SCALAR*,       XL, Kokkos::Device<Kokkos::Cuda, MEMSPACE>, Kokkos::MemoryTraits<Kokkos::Unmanaged | Kokkos::RandomAccess>, \
+			   SCALAR*,             YL, Kokkos::Device<Kokkos::Cuda, MEMSPACE>, Kokkos::MemoryTraits<Kokkos::Unmanaged> > { \
+  enum : bool { value = true }; \
+};
+#endif // KOKKOSKERNELS_ENABLE_TPL_ROCSPARSE
+
```

Comment:
I think the tpl_spec_avail changes shouldn't be merged in until there's the corresponding wrapper in tpl_spec_decl.

Reply:
Agreed, I will get that done, I forgot I had that change in `tpl_spec_avail`